Replace Nodemailer with SendGrid API to solve SMTP blocking issues on Digital Ocean

IrisWRX opened the pull request:
This pr replaces the existing Nodemailer email implementation with SendGrid for more reliable email delivery. Additionally, it implements a two-step email verification process to solve email client prefetching issues, particularly with institutional email providers like BCIT.

## Code changes
- Added `@sendgrid/mail` package
- Created `sendgrid.middleware.ts` to handle email sending
- Updated user model
- Added new API endpoints for verification: `check-verification` and `confirm-verification`
- Updated verification logic 
- Added verification button requiring user interaction
- Updated routing to handle query parameters for verification

## To test
Now we can test verifying new users, forgetting passwords, and resending verification emails in deployed environments. (Please also use your bcit school email to test)

IrisWRX committed:
Replaced Nodemailer with SendGrid API to solve SMTP blocking issues on Digital Ocean

IrisWRX commented:
Need to make some changes.

IrisWRX committed:
Fixed email verification to handle email client link prefetching issue

IrisWRX commented:
Added a fix

IrisWRX commented:
Link prefetching issue remains for bcit email

IrisWRX committed:
Fix school email prefetch issue during email verification

IrisWRX commented:
The school email prefetch issue has been handled now.

JoanneH2050 commented:
was able to test new user verification :D
email comes from noreply@iris-xu.me for now
resend verification btn throws Message.EmailAddressRequired error
https://www.loom.com/share/99c6521bb3b74217a7264d5e714c5911?sid=ab96dffa-36f7-4846-9406-dee304f23c07

IrisWRX committed:
Fixed the bug of not being able to resend verification email

IrisWRX commented:
@JoanneH2050 Thanks for identifying the issues! I've updated the SendGrid API key and the sender email in the deployment .env file, so you should now receive emails from `noreply@mytechie.pro`. I've also fixed the `Message.EmailAddressRequired` error. Now we should be able to resend the verification email. Could you test these again on the deployed environment? Thank you:)
